api/src/main/kotlin/com/openlattice/organization/OrganizationExternalDatabaseView.kt

```diff
@@ -0,0 +1,69 @@
+package com.openlattice.organization
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.openlattice.authorization.securable.AbstractSecurableObject
+import com.openlattice.authorization.securable.SecurableObjectType
+import com.openlattice.client.serialization.SerializationConstants
+import java.util.*
+
+/**
+ * Creates a securable object for an organization's entire database
+ *
+ * @param id An optional UUID that will be automatically generated if not provided
+ * @param name The name of the table
+ * @param title A title for the object
+ * @param description An optional description of this object
+ * @param organizationId The id of the organization that owns this table
+ */
+
+class OrganizationExternalDatabaseView
+
+constructor(
+        @JsonProperty(SerializationConstants.ID_FIELD) id: Optional<UUID>,
+        @JsonProperty(SerializationConstants.NAME_FIELD) var name: String,
+        @JsonProperty(SerializationConstants.TITLE_FIELD) title: String,
+        @JsonProperty(SerializationConstants.DESCRIPTION_FIELD) description: Optional<String>,
+        @JsonProperty(SerializationConstants.ORGANIZATION_ID) var organizationId: UUID,
+        @JsonProperty(SerializationConstants.DATA_SOURCE_ID) val dataSourceId: UUID,
+        @JsonProperty(SerializationConstants.EXTERNAL_ID) val externalId: String
+) : AbstractSecurableObject(id, title, description) {
```

Comment:
Looks like it -- anyway it might make sense to have some AbstractDatabaseSecurableObject or something that contains these overlapping fields just to avoid a lot of duplication -- although we don't really have anything in place for the base objects to simplify StreamSerializers, ResultSetAdapters, Mapstores, etc, so now that I think about it it probably wouldn't end up avoiding much duplication at all.

(although it *would* be nice to have something like that someday for simplification/consistency...)

Reply:
All these objects are intermediate term hacks and I wouldn't worry too much about them or their manipulation just yet. These are all duplicated into entity sets and we need to rationalize where that data will live and how it will be read.

Reply:
Broadly you should ignore OrganizationExternal* for now.